### PR TITLE
fix: weaviate patched image with fixed CVEs

### PR DESCRIPTION
**Pull request by SandhyaRavi2403**
**What problem does this PR solve?**:
patch weaviate image with fixed CVE's

`sandhya.ravi@GT9X7CVF5F kommander-applications % trivy image ghcr.io/mesosphere/dkp-container-images/cr.weaviate.io/semitechnologies/weaviate:1.21.4-d2iq.0
--vuln-type os --ignore-unfixed | grep Total
2024-07-16T00:23:52+05:30 INFO Need to update DB
2024-07-16T00:23:52+05:30 INFO Downloading DB... repository="ghcr.io/aquasecurity/trivy-db:2"
49.73 MiB / 49.73 MiB [---------------------------------------------------------------------------------------] 100.00% 3.46 MiB p/s 15s
2024-07-16T00:24:08+05:30 INFO Vulnerability scanning is enabled
2024-07-16T00:24:08+05:30 INFO Secret scanning is enabled
2024-07-16T00:24:08+05:30 INFO If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-07-16T00:24:08+05:30 INFO Please see also https://aquasecurity.github.io/trivy/v0.53/docs/scanner/secret#recommendation for faster secret detection
2024-07-16T00:24:11+05:30 INFO Detected OS family="alpine" version="3.18.3"
2024-07-16T00:24:11+05:30 INFO [alpine] Detecting vulnerabilities... os_version="3.18" repository="3.18" pkg_num=17
2024-07-16T00:24:11+05:30 INFO Number of language-specific files num=1
2024-07-16T00:24:11+05:30 INFO [gobinary] Detecting vulnerabilities...
2024-07-16T00:24:11+05:30 WARN Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.53/docs/scanner/vulnerability#severity-selection for details.

ghcr.io/mesosphere/dkp-container-images/cr.weaviate.io/semitechnologies/weaviate:1.21.4-d2iq.0 (alpine 3.18.3)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)`

workflow link:
https://github.com/mesosphere/dkp-container-images/actions/runs/9844219091

**Which issue(s) does this PR fix?**:

https://jira.nutanix.com/browse/NCN-101394

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

**Comment by SandhyaRavi2403** [comment]
corresponding kommander PR: https://github.com/mesosphere/kommander/pull/4860

**Comment by SandhyaRavi2403** [comment]
image updated in ai-navigator repo - https://github.com/mesosphere/ai-navigator-cluster-info-api/pull/29

**Comment by SandhyaRavi2403** [comment]
This PR is updated with the changes from postgresql https://github.com/mesosphere/kommander-applications/pull/2403/files

**Comment by SandhyaRavi2403** [comment]
The corresponding PR's
**ai-navigator:** https://github.com/mesosphere/ai-navigator-cluster-info-api/pull/14
**dkp-insights**: https://github.com/mesosphere/dkp-insights/pull/1515 (merged)

File: services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml [hunk 1]
```diff
@@ -23,4 +23,4 @@ spec:
       retries: 30
   valuesFrom:
     - kind: ConfigMap
```
Comment (lukeogg) [inline services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml:25]: `version: 0.2.1` needs to be bumped, and a new chart needs to be released for the `values.yaml` update. It's manual right now. I'll release `version: 0.2.2` Please bump to that.

**Review by lukeogg — CHANGES_REQUESTED** [review]

**Review by lukeogg — APPROVED** [review]

File: services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml (hunk 1 above)
Comment (lukeogg) [reply services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml:25]: > `version: 0.2.1` needs to be bumped, and a new chart needs to be released for the `values.yaml` update. It's manual right now. I'll release `version: 0.2.2` Please bump to that.

Bumped.

Comment (lukeogg) [reply services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml:25]: Another inconsistency that was preventing me from updating the license file:

This PR has `ghcr.io/mesosphere/dkp-container-images/cr.weaviate.io/semitechnologies/weaviate:1.21.4`
the [AI Nav API now has](https://github.com/mesosphere/ai-navigator-cluster-info-api/blob/5b16b4698e304b4e15c635684146a5d26e2d264b/chart/cluster-info-api/values.yaml#L29-L31) `ghcr.io/mesosphere/dkp-container-images/cr.weaviate.io/semitechnologies/weaviate:1.21.4-d2iq.0`

Which one is it? These need to be consistent, and the license file needs to be updated.

Same problem with `ghcr.io/mesosphere/dkp-container-images/docker.io/bitnami/postgresql:15.2.0-debian-11-r21` and `ghcr.io/mesosphere/dkp-container-images/docker.io/bitnami/postgresql:15.2.0-debian-11-r21-d2iq.0`

Comment (mhrabovcin) [reply services/ai-navigator-app/0.2.4/helmrelease/helmrelease.yaml:25]: Great catch, there was the `tag` value in the defaults CM which was overriding helm default.

```
    # Default values for subcharts
    weaviate:
      enabled: true
      globalPriorityClassName: dkp-high-priority
      service:
        type: ClusterIP
      image:
        tag: 1.21.4
```

**Review by lukeogg — APPROVED** [review]
LGTM!

**Pull request merged**